archiver.go

```diff
@@ -168,32 +171,120 @@ select rec.visibility, row_to_json(rec) FROM (
 		ELSE NULL
 		END as visibility,
 	  text,
-	  (select coalesce(jsonb_agg(attach_row), '[]'::jsonb) FROM (select attach_data.attachment[1] as content_type, attach_data.attachment[2] as url FROM (select regexp_matches(unnest(attachments), '^(.*?);(.*)$') attachment FROM msgs_msg where id = 8) as attach_data) as attach_row) as attachments,
+	  (select coalesce(jsonb_agg(attach_row), '[]'::jsonb) FROM (select attach_data.attachment[1] as content_type, attach_data.attachment[2] as url FROM (select regexp_matches(unnest(attachments), '^(.*?);(.*)$') attachment) as attach_data) as attach_row) as attachments,
 	  labels_agg.data as labels,
-	  created_on,
+	  mm.created_on,
 	  sent_on
-	from msgs_msg mm JOIN contacts_contacturn ccu ON mm.contact_urn_id = ccu.id
+	from msgs_msg mm JOIN contacts_contacturn ccu ON mm.contact_urn_id = ccu.id JOIN orgs_org oo ON ccu.org_id = oo.id
 	  JOIN LATERAL (select uuid, name from contacts_contact cc where cc.id = mm.contact_id) as contact ON True
 	  JOIN LATERAL (select uuid, name from channels_channel ch where ch.id = mm.channel_id) as channel ON True
 	  LEFT JOIN LATERAL (select coalesce(jsonb_agg(label_row), '[]'::jsonb) as data from (select uuid, name from msgs_label ml INNER JOIN msgs_msg_labels mml ON ml.id = mml.label_id AND mml.msg_id = mm.id) as label_row) as labels_agg ON True
-	
-	
-	  WHERE mm.org_id = $1 AND mm.modified_on >= $2 AND mm.modified_on < $3
-	order by modified_on ASC, id ASC) rec; 
+
+	  WHERE mm.org_id = $1 AND mm.created_on >= $2 AND mm.created_on < $3
+	order by created_on ASC, id ASC) rec; 
 `
 
-func CreateMsgArchive(ctx context.Context, db *sqlx.DB, task *ArchiveTask) error {
-	task.BuildStart = time.Now()
+const lookupFlowRuns = `
+select row_to_json(rec)
+FROM (
+   select
+     fr.id,
+     row_to_json(flow_struct) as flow,
+     row_to_json(contact_struct) as contact,
+     fr.responded,
+     (select jsonb_agg(path_data) from (
+          select path_row ->> 'node_uuid'  as node, path_row ->> 'arrived_on' as time
+          from jsonb_array_elements(fr.path :: jsonb) as path_row) as path_data
+     ) as path,
+     (select jsonb_agg(values_data.tmp_values) from (
+          select json_build_object(key, jsonb_build_object('name', value -> 'name', 'time', value -> 'created_on', 'category', value -> 'category', 'node', value -> 'node_uuid')) as tmp_values
+          FROM jsonb_each(fr.results :: jsonb)) as values_data
+     ) as values,
```

Comment:
@rowanseymour so do you think we should throw in events here as well? And if so, just in the raw form they are in?